packages/flame/lib/src/components/clip_component.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+
+/// {@template clip_component}
+/// A component that will clip its content.
+/// {@endtemplate}
+abstract class ClipComponent extends PositionComponent {
+  /// {@macro clip_component}
+  ///
+  /// Clips the canvas based its shape and size.
+  ClipComponent({
+    super.position,
+    super.size,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+  });
+
+  late Path _path;
+  late Shape _shape;
+
+  @override
+  Future<void> onLoad() async {
+    _prepare();
+    size.addListener(_prepare);
```

Comment:
size has a single instance internally https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/components/position_component.dart#L178

I don't see the point of exposing this implementation detail to the user, if the size never change, yes the listener is added in vein, but there is no harm in that since this is an internal thing.

Reply:
~~in vein~~ in vain.
A simpler approach could be to just override the `size` setter -- this way whenever the user changes the size, the shape gets updated too. This would be slightly less general, in case the user changes the size vector in-place, but on the other hand such practice ought to be discouraged anyway.

Reply:
I honestly don't think that the listener will impact anything, at most this will allocate something in the memory that may not be used.

But I would argue that this memory footprint will be marginal and this has no risk of leak, and the alternative is weird make it harder for the user, or to "hack" by overriding methods or other ways.